pkg/rancher-desktop/backend/wsl.ts

```diff
@@ -1266,7 +1268,13 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
 
                 await this.execCommand({ root: true }, 'rm', '-f', obsoleteIALConfFile);
               }),
-              this.progressTracker.action('Rancher Desktop guest agent', 50, this.installGuestAgent(kubernetesVersion, this.cfg)),
+              this.progressTracker.action('Rancher Desktop guest agent', 50, async() => {
+                // we do not want to run the guest agent for the new rancher desktop networking
+                // since this can cause port forwarding duplication and clobber the current process.
+                if (!this.cfg?.experimental.virtualMachine.networkingTunnel) {
+                  await this.installGuestAgent(kubernetesVersion, this.cfg);
+                }
+              }),
```

Comment:
Wouldn't it make more sense the other way around?
```ts
if (!this.cfg?.experimental.virtualMachine.networkingTunnel) {
  this.progressTracker.action(…
```
That way we can never accidentally show (like a single frame of) "Rancher Desktop guest agent".  (In practice, the priority here isn't high enough for that to happen anyway.)

Reply:
I agree that makes more sense but this progress tracker action is part of a `promise.all<T>(values: Iterable<T | PromiseLike<T>>)...` which is an array of values. I wasn't 100% sure on how to squeeze an if block in there if there is a way I'm happy to change it. Thanks